MagazineLayout/Public/UICollectionViewDelegateMagazineLayout.swift

```diff
@@ -135,4 +137,247 @@ public protocol UICollectionViewDelegateMagazineLayout: UICollectionViewDelegate
     insetsForItemsInSectionAtIndex index: Int)
     -> UIEdgeInsets
 
+  ///   Asks the delegate to modify a layout attributes instance so that it represents the initial visual state of an item being inserted via
+  ///   `UICollectionView.insertItems(at:)`.
+  ///
+  ///   The `initialLayoutAttributes` instance is a reference type, and therefore can be modified directly. If the provided
+  ///   layout attributes instance is not changed in the implementation of this function, then the item will be inserted with no animation.
+  ///
+  ///   - Parameters:
+  ///      - collectionView: The collection view using the layout.
+  ///      - collectionViewLayout: The layout requesting the information.
+  ///      - indexPath: The index path of the item.
+  ///      - initialLayoutAttributes: The unmodified layout attributes representing the final visual state of the inserted
+  ///      item. Modify properties on this instance to create an item insert animation.
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    initialLayoutAttributesForInsertedItemAt indexPath: IndexPath,
+    byModifying initialLayoutAttributes: UICollectionViewLayoutAttributes)
```

Comment:
While it's a bit unconventional to pass in a reference and expect the delegate conformer to mutate it (is this done anywhere else in UIKit?), I think this API makes it pretty straightforward to do the right thing.

- There's no return value, so there's nothing that would encourage an API implementer to make a `copy` of the layout attributes, modify that, and return that
- The naming straight up tells you to modify the provided layout attributes (`byModifying`)
- The documentation mentions, multiple times, that you should mutate the provided layout attributes directly, and also tells you what will happen if you don't modify the provided layout attributes

If somebody still manages to mess it up, then they don't deserve pretty animations 😆

Reply:
With all of that said, there's an alternative API:

API implementers could instead provide some struct that encapsulates all of the layout attributes properties, then internally in the layout, I'd apply these properties from the provided struct to the layout attributes.

Reply:
Downsides to this API:
- I need to make sure it's always kept up-to-date with whatever properties `UICollectionViewLayoutAttributes` has

In practice, that may equate to 0 effort

Reply:
> While it's a bit unconventional to pass in a reference and expect the delegate conformer to mutate it (is this done anywhere else in UIKit?)

It's done [here](https://developer.apple.com/documentation/uikit/uiscrollviewdelegate/1619385-scrollviewwillenddragging), which I guess is so that you can go with the system provided offset if you choose.
```
func scrollViewWillEndDragging(_ scrollView: UIScrollView, 
                           withVelocity velocity: CGPoint, 
                    targetContentOffset: UnsafeMutablePointer<CGPoint>)
```

Reply:
I agree that this current API is clear enough though, and doesn't have any significant drawbacks.